test/state/state.cpp

```diff
@@ -53,24 +53,34 @@ int64_t compute_tx_intrinsic_cost(evmc_revision rev, const Transaction& tx) noex
 }
 
 /// Validates transaction and computes its execution gas limit (the amount of gas provided to EVM).
-/// @return  Non-negative execution gas limit for valid transaction
-///          or negative value for invalid transaction.
+/// @return  Execution gas limit or transaction validation error.
 std::variant<int64_t, std::error_code> validate_transaction(const Account& sender_acc,
     const BlockInfo& block, const Transaction& tx, evmc_revision rev) noexcept
 {
-    if (rev < EVMC_LONDON && tx.kind == Transaction::Kind::eip1559)
-        return make_error_code(TX_TYPE_NOT_SUPPORTED);
+    switch (tx.kind)
```

Comment:
Is the goal of using switch here to hope for compiler warnings when we get non-handled types at some point?
(otherwise it's quite peculiar switch with two fallthroughs and one empty case)

Reply:
The goal was to group checks by transaction type so it is more visible which transaction type introduce a particular check.
The blob transactions will extend it by 4 more checks. https://github.com/ethereum/evmone/pull/668/commits/336e0f9fc08f1f27c9bb26e5330c8202afefcd44#diff-51a631fbac5b8dec0cfc6a2d9dcb58a01eedc369ac3730150d40f4eaa4f9c1e3R88-R98